daemon/containerd/image_exporter.go

```diff
@@ -248,8 +248,21 @@ func (i *ImageService) LoadImage(ctx context.Context, inTar io.ReadCloser, platf
 			if nameFromArchive == "" {
 				return false
 			}
-			_, err := reference.ParseNormalizedNamed(nameFromArchive)
-			return err == nil
+
+			ref, err := reference.ParseNormalizedNamed(nameFromArchive)
+			if err != nil {
+				return false
+			}
+
+			// Look up if there is an existing image with this name and ensure a dangling image exists.
+			if img, err := i.resolveImage(ctx, ref.String()); err != nil && !errdefs.IsNotFound(err) {
+				log.G(ctx).WithError(err).Warn("failed to resolve image: %w", err)
+			} else {
+				if err := i.ensureDanglingImage(ctx, img); err != nil {
```

Comment:
Plain `else` would also match the `if err != nil && errdefs.IsNotFound(err)` case - which will create a dangling image if the image is not found.

(that will be solved by applying the previous suggestion 😅)

Reply:
I fixed this so it only calls `ensureDanglingImage` when there's no error. I changed the logic so the `else` comes first and the warning message about being unable to find the image happens in the else because it made the logic more clear.